app/src/org/commcare/views/Combobox.java

```diff
@@ -0,0 +1,138 @@
+package org.commcare.views;
+
+import android.content.Context;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.TypedValue;
+import android.view.View;
+import android.widget.AutoCompleteTextView;
+
+import org.commcare.adapters.ComboboxAdapter;
+import org.commcare.adapters.PermissiveComboboxAdapter;
+import org.commcare.dalvik.R;
+import org.commcare.views.widgets.SpinnerWidget;
+
+import java.util.Vector;
+
+/**
+ * Custom view that builds upon Android's built-in AutoCompleteTextView and adds:
+ * -Validation and auto-correcting of user-entered text to only allow strings that are part of
+ * an available answer choice
+ * -Formatting to make the drop-down view look identical to that of CommCare's SpinnerWidget
+ * -The ability to add a custom adapter that implements custom filtering rules (rather than using
+ * the default filter of AutoCompleteTextView)
+ *
+ * @author Aliza Stone
+ */
+public class Combobox extends AutoCompleteTextView {
+
+    private Vector<String> choices;
+    private CharSequence lastAcceptableStringEntered = "";
+    private int lastValidCursorLocation;
+    private boolean fixingInvalidEntry;
+    private ComboboxAdapter customAdapter;
+
+    public Combobox(Context context, Vector<String> choices, boolean addEmptyFirstChoice,
+                    boolean permissive, int fontSize) {
```

Comment:
@ctsims So the reason I did this is because I was copying what `SpinnerWidget` does here: https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/views/widgets/SpinnerWidget.java#L52. It ends up looking pretty bad to just use the AutoCompleteTextView's native font size, because it's way smaller than the other text on the screen. But I agree that it looks out-of-place from the perspective of the `Combobox` view's code, which doesn't know anything about widget world. Is it sufficient in your mind to change the constructor to not take in font size, and then add a static constructor method called like `createComboboxForUseInWidget()` that does set the font size?